### ISSUE TITLE
V2 Support

### ISSUE THREAD

**Issue by keliam** [ryanbdclark/owlet#8]
Are you still up for adding support for the V2 sock? I have a V2 and would be happy to offer any information to get it up and running.

**Comment by ryanbdclark** [comment]
Yep definitely, I'll send you some details later today about the info I need

**Comment by ryanbdclark** [comment]
@keliam if you are able to follow the steps below and upload the resulting
 json file that would be great then I can work to integrate V2 socks.

I have quickly created the below script from my pyowletapi library that will authenticate against the Owlet API, get the properties JSON from the Owlet API and then save it to a file
[get_owlet_properties.zip](https://github.com/ryanbdclark/owlet/files/13366978/get_owlet_properties.zip)

To run extract the python file and open a command line/terminal at the same location this python file is saved (mine was just on the desktop)
![image](https://github.com/ryanbdclark/owlet/assets/42173121/18e47efd-1d47-415d-8201-848741f19bc7)

Run the script using the below command: 

python get_owlet_properties.py _region_ _email_ _password_

region should be replaced with either europe or world depending on where you are based
email should be replaced with the email you signed up to owlet with
password should be replaced with your owlet account email

This will do the necessary API calls and then simply say complete when done. You should find that there is now a properties.json file in the same location the python file is saved. If you could upload that file that would be great, ideally this would be done when the sock is in use but if not then that's also fine. 

I'm hoping that with the V2 sock everything on the API is exactly the same except the output will be slightly different but if you do get any issues when running the script if you let me know of any errors you get then reply with them here and I can have a look.

**Comment by keliam** [comment]
Not sure if I did something wrong. I'm in the file location, double checked spellings/case and information.

`ModuleNotFoundError: No module named 'requests'`

**Comment by ryanbdclark** [comment]
Apologies, I thought requests was a built in module. You’ll need to install it first by running the below command in a terminal. 
`python -m pip install requests`

**Comment by keliam** [comment]
Thanks - not familiar with python whatsoever.

Here is the result.

[properties.json](https://github.com/ryanbdclark/owlet/files/13415666/properties.json)

**Comment by ryanbdclark** [comment]
Thanks for providing that, I will have a look at this in the next few days and try and push out an update for V2 support this week. If I need anything else I'll let you know

**Comment by ryanbdclark** [comment]
@keliam I've just released version 2023.11.2 which now has V2 support!

I've tested as much as I can although without an actual V2 sock I obviously can't do real world testing. 

Let me know how you get on with setting the integration up. I've done my best to interpret the values from the JSON you provided but if anything doesn't look right then let me know.

Please let me know if everything is ok so I can close this issue off.

**Comment by keliam** [comment]
Awesome, thanks! The sensors look like they’re coming through okay so far. I’ll just need to double check once it’s in use tonight.

**Comment by keliam** [comment]
Everything looks good. The only issue is that the normal iOS app seems to not have any history for last night, which is the first night I had the integration pulling data. Any resolution to that?

**Comment by ryanbdclark** [comment]
Is the history issue still occurring, the integration shouldn't have an impact on the history. I know that the history on my app still shows ok and have been running the integration for many months now. If it's still occurring are you able to disable/delete the integration to see if it starts working again without the integration? If this is caused by the integration I'm not sure there is anything I could do to resolve this as all the integration is doing is calling the API for data.

**Comment by keliam** [comment]
It seems the issue may have been resolved by an action that Owlet support offered. I do think it happened when the integration was added, but I'm not sure the reason why. I added a different V2/V3 Owlet integration previously and it happened then as well. When I found your integration I was hoping the issue wouldn't be present. Luckily the solution isn't related to either integration, so I think I'll be good to go from here on out. In case anyone else has the issue, here is what Owlet suggests trying.
- Create new/dummy account
- Move the sock/device to the newly created account
- Move device back to your primary account
- Feel free to delete dummy account

Thanks for getting V2 added! I'll let you know if any issues arise.
